File: test/workflowsuite/kpi/client/bits/BitConverterTest.java
package workflowsuite.kpi.client.bits;

import org.junit.Assert;
import org.junit.Test;

public final class BitConverterTest {
    @Test
    public void byteArrayToLongTest() {
        long expected = 4676137844535450131L;
        byte[] bytes = new byte[] {19, (byte)218, 75, 40, (byte)152, (byte)250, (byte)228, 64};

        long actual = BitConverter.byteArrayToLongLE(bytes, 0);
        Assert.assertEquals(expected, actual);
    }

    @Test
    public void byteArrayToDoubleTest() {
        double expected = 42964.754918981482;
        byte[] bytes = new byte[] {19, (byte) 218, 75, 40, (byte) 152, (byte) 250, (byte) 228, 64};

        double actual = BitConverter.byteArrayToDoubleLE(bytes, 0);
        Assert.assertEquals(expected, actual, 0.0);
    }
}

File: test/workflowSuite/kpi/client/bits/BitConverterTest.kt
package workflowsuite.kpi.client.bits

import org.junit.Assert
import org.junit.Test

class BitConverterTest {
    @Test
    fun byteArrayToLongTest() {
        val expected = 4676137844535450131
        val bytes = byteArrayOf(19, 218.toByte(), 75, 40, 152.toByte(), 250.toByte(), 228.toByte(), 64)

        val actual = BitConverter.byteArrayToLongLE(bytes, 0)
        Assert.assertEquals(expected, actual)
    }

    @Test
    fun byteArrayToDoubleTest() {
        val expected = 42964.754918981482
        val bytes = byteArrayOf(19, 218.toByte(), 75, 40, 152.toByte(), 250.toByte(), 228.toByte(), 64)

        val actual = BitConverter.byteArrayToDoubleLE(bytes, 0)
        Assert.assertEquals(expected, actual, 0.0)
    }
}